lib/Default/SmrWeapon.class.inc

```diff
@@ -249,44 +245,52 @@ class SmrWeapon extends AbstractSmrCombatWeapon {
 		return $damage;
 	}
 	
+	public function checkHit(AbstractSmrPlayer $player, $weaponAccuracy) {
+		$this->getWeightedRandomForPlayer($player)->flipWeightedCoin($weaponAccuracy);
```

Comment:
This is missing a return value. If you used this code in testing the changes, then you probably should have seen misses 100% of the time.